index.d.ts

```diff
@@ -0,0 +1,39 @@
+export interface AuthConfiguration extends BaseAuthConfiguration {
+    scopes: string[];
+    redirectUrl: string;
+    aditionalParameters?: {[name: string]: string};
+  }
+
+  export interface BaseAuthConfiguration{
```

Comment:
can you move this to the top as `AuthConfiguration` extends it before it's being defined. Just for readability :+1: